src/main/java/sirius/biz/jobs/batch/file/FileImportJob.java

```diff
@@ -42,11 +51,68 @@ protected FileImportJob(VirtualObjectParameter fileParameter, ProcessContext pro
     @Override
     public void execute() throws Exception {
         VirtualObject file = process.require(fileParameter);
+
         try (InputStream in = storage.getData(file)) {
+            if (!canHandleFileExtension(file.getFileExtension()) && "zip".equalsIgnoreCase(file.getFileExtension())) {
+                executeForZippedFile(in);
+
+                return;
+            }
+
+            if (!canHandleFileExtension(file.getFileExtension())) {
```

Comment:
combine with the if above and also without loading the data first (above storage.getData())